### DISCUSSION TITLE
Incremental counters and prometheus

### DISCUSSION THREAD

**Discussion by mrblonde91** [open-telemetry/opentelemetry-dotnet#4440] [Q&A]
So we've got incremental counters set up to track a number that maybe  increases a dozen times in one go and then remains largely unchanged a lot of time. When this reaches prometheus, it invariably seems to get lost, either reporting no hits or misreporting with a way lower number. Is this by virtue of absolute accuracy not being possible for such small and rare hits?

**Comment by cijothomas** [comment]
Not sure If I understand the issue. Could you elaborate with a minimal example? Particularly the part about _"seems to get lost"_

**Reply by mrblonde91** [reply]
So simple one, we might add 7 to a count within a minute but nothing further for maybe an hour. This doesn't appear to be reflected when we query Prometheus for the data, we might see no numbers or a fraction of reality. 

For higher frequency counters, they're pretty accurate but low frequency, it seems to be less so. Confirmed the inaccuracy via logs.

**Reply by cijothomas** [reply]
> This doesn't appear to be reflected when we query Prometheus for the data

Can you show a minimal repro app? I doubt if this has anything to do with OpenTelemetry Prometheus Exporter. (It currently never clears any instruments unless user explicitly disposes the Meter. So irrespective of if you are actively adding more counter.Add()s, the value will always be reported to Prometheus.
